lib/postprocessing/image.js

```diff
@@ -94,113 +93,50 @@ function processBasicImageOperations(img, instructions) {
+async function imagePostProcess(infile, outfile, instructions) {
     await logImageMagickVersion();
 
-    return new Promise(function (resolve, reject) {
-        const outdir = path.dirname(outfile);
-        const renditionName = path.basename(outfile);
-
-        // for pdfs and multi-page docs/images, take the first one only
-        let img = gm(`${infile}[0]`);
-
-        // const selectedOperations = operations || DEFAULT_OPERATIONS;
-        // turning off watermarking for increamental post-processing
-        const selectedOperations = DEFAULT_OPERATIONS;
-        img = applyImageOperations(img, instructions, selectedOperations);
-
-        img.write(outfile,
-            function (err) {
-                if (err) {
-                    console.log("FAILURE of rendition processing using image processing lib (imagemagick)", renditionName);
-                    console.error("failed conversion:", err);
-                    reject(`conversion using image processing lib (imagemagick) failed: ${err.message}, code: ${err.code}, signal: ${err.signal}`);
-                } else {
-                    // If we specified jpegSize for JPEG output we need to verify that the size does not exceed that
-                    const size = fileSize(outfile);
-                    if (instructions.fmt === "jpg" && instructions.jpegSize && size > instructions.jpegSize) {
-                        const newImg = gm(outfile);
-
-                        // From http://www.imagemagick.org/Usage/formats/#jpg
-                        // As of IM v6.5.8-2 you can specify a maximum output filesize for the JPEG image.
-                        newImg.define(`jpeg:extent=${instructions.jpegSize}`);
-                        const tmpFile = `${outdir}/${tempFileName("jpg")}`;
-                        newImg.write(tmpFile,
-                            function (err) {
-                                if (err) {
-                                    console.log("FAILURE of rendition processing (fmt=jpg) using image processing lib (imagemagick)", renditionName);
-                                    console.error("failed conversion:", err);
-                                    reject(`cconversion using image processing lib (imagemagick) failed: ${err.message}, code: ${err.code}, signal: ${err.signal}`);
-                                } else {
-                                    fs.renameSync(tmpFile, outfile);
-                                    console.log("END of rendition processing using image processing lib (imagemagick)", renditionName);
-                                    resolve(true);
-                                }
-                            });
-                    } else {
-                        console.log("END of rendition processing using image processing lib (imagemagick)", renditionName);
-                        resolve(true);
-                    }
-                }
-            });
-    });
+    const img = await read(infile);
```

Comment:
Not needed for now, but I would add an index as optional parameter to read.
This would be useful if a SDK user wants something else than the first picture from "something".

Reply:
Yeah, we have this `[0]` inside the `read()` there from the inherited code from worker-imagemagick. Not sure if it's actually used since we have dedicated workers for pdf and other document formats.

I think we can tackle it when we have more requirements, as it might also be an api addition that I wouldn't want to rush before we know exactly what we need.

Reply:
The indexing is to select the first image in multipage files - like pdf. And it was used in the pdf worker to select the first page of the pdf (that changed now in another PR), in order to only work on that first-page image.